### ISSUE TITLE
Issue when using element queries in Firefox with Typekit

### ISSUE THREAD

**Issue by fchristant** [marcj/css-element-queries#117]
First of all, amazing library!

I'd like to report a pretty specific issue where element queries fails for me in Firefox when using Typekit. Here's the relevant part of my head section:

```
    <link rel="stylesheet" type="text/css" href="<?= $base ?>assets/css/main.css" crossorigin="anonymous" />

    <script src="../src/js/head/ResizeSensor.js"></script>
    <script src="../src/js/head/ElementQueries.js"></script>

    <script src="//use.typekit.net/lib.js"></script>
    <script>try{Typekit.load({async:true});}catch(e){}</script>

    <link rel="stylesheet" type="text/css" href="http://fonts.googleapis.com/css?family=Lobster&&effect=canvas-print" crossorigin="anonymous">
```

I've anonymized the above code somewhat, since it uses a paid typekit subscription. I'm getting this error in the Firefox console:

**InvalidAccessError: A parameter or an operation is not supported by the underlying object**

This would be a typical CORS error when the library tries to read CSS rules from an external stylesheet. I've narrowed it down to the typekit stylesheet throwing the issue, which is a CSS that is dynamically fetched by the typekit JS. For the record, the response from typekit does allow all domains, so it is strange that the error occurs in the first place. It's equally strange that the Google web fonts request does not have this issue. 

I'm fine though that the typekit CSS rules are not read, the problem is that it blocks the working of the whole EQ lib altogether. I basically want EQ to apply to my main.css only, and don't care about the failure in reading any other CSS files. 

Is there a way to do that? Or perhaps a way for the lib to be more robust in that when it fails, it just moves on to the next CSS with success?

**Comment by jyrkij** [comment]
Hacky solution: Change [line #389 of ElementQueries.js](https://github.com/marcj/css-element-queries/blob/6c125e79b226b62f6fa930e4debd563c1f9f89df/src/ElementQueries.js#L389) from

```js
                    if (e.name !== 'SecurityError') {
```

to

```js
                    if (e.name !== 'SecurityError' && e.name !== 'InvalidAccessError') {
```

This ignores the error...

**Comment by marcj** [comment]
@jyrkij care to send a PR?

**Comment by jyrkij** [comment]
@marcj any idea if this'll suppress some other errors we might want to throw?

**Comment by marcj** [comment]
I guess we can silently ignore those errors.

**Comment by Ephys** [comment]
`InvalidAccessError` can be (or are ?) caused by CORS problems, it could be interesting to warn the developer that one of their css files is unreadable and let them decide whether or not their app can continue without it.

My proposed solution would be to add a mechanism to listen for errors on ElemQueries and dispatch these errors rather than ignoring them. I would also recommend logging the error (only if there is no error listener ?) so the developers don't think that their layout being broken is ElemQueries' fault.

**Comment by marcj** [comment]
Logging means you display always a log entry also for cases where you can't fix it (like with font services), which is not so sexy. Maybe a dev mode makes sense.

**Comment by mandalatv** [comment]
Was a PR ever approved for @jyrkij's fix? I ran into the same problem and his solution resolved it.

**Comment by jyrkij** [comment]
Well, I never got to making a pull request as I thought there wasn't any conclusion on how to implement the fix. Should we just silence them (as my change does) or should there be a dev mode or logging or...

**Comment by mandalatv** [comment]
What about doing a console.error()? That has more weight than a simple log but doesn't prevent rendering in Firefox. 

FYI (I just saw this issue's title, again), for the record, I'm not using TypeKit but am getting a lack of rendering in Firefox 54.0a2 (2017-04-17) (64-bit). My app is calling an API that populates content into child containers within the element, so it's almost definitely a CORS issue.

**Comment by marcj** [comment]
Fixed in upcoming 1.0.0 version.
